python/modem-firmware-1.3+/device_credentials_installer.py

```diff
@@ -645,10 +647,10 @@ def main():
 
     cred_if = None
     if args.cmd_type == CMD_TYPE_AT:
-        cred_if = ATCommandInterface(write_line, wait_for_prompt, verbose)
+        cred_if = ATCommandInterface(write_line, wait_for_prompt, verbose, args.shell)
 
     if args.cmd_type == CMD_TYPE_TLS_SHELL:
-        cred_if = TLSCredShellInterface(write_line, wait_for_prompt, verbose)
+        cred_if = TLSCredShellInterface(write_line, wait_for_prompt, verbose, True)
```

Comment:
```suggestion
  if args.cmd_type == CMD_TYPE_AT:
        cred_if = ATCommandInterface(write_line, wait_for_prompt, verbose, args.shell)

  if args.cmd_type == CMD_TYPE_AT_SHELL:
      cred_if = ATCommandInterface(write_line, wait_for_prompt, verbose, args.shell)
      cred_if.set_shell(True)

    if args.cmd_type == CMD_TYPE_TLS_SHELL:
        cred_if = TLSCredShellInterface(write_line, wait_for_prompt, verbose, True)
 ```

Reply:
Forgive me for the awful indenting, github is not the best editor

Reply:
I did it differently because DRY